feat: dev환경 terraform 구성 및 CI/CD 파이프라인 설계

yongjun0511 opened the pull request:
## 🌱 관련 이슈
- close #7 

---
## 📌 작업 내용 및 특이사항
- prod과 dev 횐경을 분리하여 배포가 이루어지도록 했습니다.
- 배포 과정에서 CI에서는 fmt과 validation 검사가 이루어지고 CD에서는 즉시 적용됩니다.
- 모듈 사용법을 README로 작성했습니다.
- dev 환경에 대한 모든 리소스 생성을 문서화했습니다
---
## 📚 참고사항
- 이제부터 aws환경에서 바뀌는 사항중 IAM을 제외한 경우 terraform을 사용해 주시면 됩니다.

yongjun0511 committed:
fix: 모듈 버전 1.12이상으로 수정

yongjun0511 committed:
chore: git ignore 설정 및 envrc 세팅

yongjun0511 committed:
refactor: 앱 이름 cherrypic 하드 코딩

yongjun0511 committed:
feat: bootstrap 구현 완료

yongjun0511 committed:
feat: bootstrap 구현 완료

yongjun0511 committed:
feat: dev 환경 terraform 구현 완료

yongjun0511 committed:
chore: 환경별 ci/cd 구현

yongjun0511 committed:
refactor: bootstrap secret.tfvars 방식으로 변경

yongjun0511 committed:
fix: count가 plan 단계에서 unknown인 문제 해결

yongjun0511 committed:
fix: Ec2 인바운드 아웃바운드 규칙 분리

yongjun0511 committed:
refactor: terraform fmt 적용

yongjun0511 committed:
fix: subnet group 인바운드 규칙 플래그 추가

yongjun0511 committed:
chore: 테라폼 포메팅 적용

yongjun0511 committed:
chore: ci/cd 파이프라인 경로 수정

ht3064 commented on .github/workflows/cd_dev.yml:
```diff
@@ -0,0 +1,43 @@
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v3
+        with:
+          aws-access-key-id: ${{ secrets.DEV_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.DEV_AWS_SECRET_ACCESS_KEY }}
```
엑세스 키 아이디와 시크릿 엑세스 키는 dev, prod 환경 분리가 필요없어 보입니다

yongjun0511 replied:
prod환경과 dev환경에서 다른 aws 서버에서 사용하는 확장성을 고려해서 설계했습니다. 동일하다면, 같은 키를 보관해도 무방합니다.

ht3064 replied:
다른 aws 계정 말씀하시는거죠? 확인했습니다

ht3064 commented on terraform/env/dev/variables.tf:
```diff
@@ -0,0 +1,22 @@
+  description = "AWS Secret Key"
+  type        = string
+  sensitive   = true
+}
```
각각 access_key_id와 secret_access_key가 적절한 선언 같습니다.

yongjun0511 replied:
수정되었습니다!

yongjun0511 committed:
chore: ec2 인스턴스 백업 ami 적용

yongjun0511 committed:
chore: 수정 사항 반영

- S3 의 lock만 사용하며, 다이나모 DB 사용하지 않기
- AWS 키 변수명 수정
- 리드미 업데이트
- Bastion host는 env를 받지 않을 수 있도록 수정
- ci에서 최신 댓글 하나만 남겨두도록 수정
- 엘라스틱 캐시는 Redis OSS로 수정

yongjun0511 committed:
chore: terraform fmt 적용

yongjun0511 committed:
fix: 오타 정정

yongjun0511 committed:
fix: 오타 정정

yongjun0511 committed:
refactor: 수정 사항 반영

- rds 기본 스키마 생성 "cherrypic_db"
- elasticache 리소스 종류 변경

yongjun0511 committed:
chore: fmt 적용

yongjun0511 committed:
chore: Ec2 README 작성

yongjun0511 committed:
chore: secret_access_key 명명 변경

yongjun0511 committed:
chore: description 수정

ht3064 reviewed (approved):
고생하셨습니다 👍
